src/rgw/rgw_rados.cc

```diff
@@ -3384,6 +3440,16 @@ class RGWRadosPutObj : public RGWHTTPStreamRWRequest::ReceiveCB
   uint64_t get_data_len() {
     return data_len;
   }
+
+  string get_calculated_etag() {
+    if (is_mpu_obj) {
+      etag_verifier_mpu->calculate_etag();
+      return etag_verifier_mpu->get_calculated_etag();
+    } else {
+      etag_verifier_atomic->calculate_etag();
```

Comment:
this will crash if it's called when rgw_copy_verify_object==false. how about:
```diff
-   } else {
+   } else if (etag_verifier_atomic) {
      etag_verifier_atomic->calculate_etag();
      return etag_verifier_atomic->get_calculated_etag();
+   } else {
+     return "";
    }
```